### ISSUE TITLE
[BUG][JavaDoc] org.opensearch.plugins package JavaDoc is missing classes

### ISSUE THREAD

**Issue by lukas-vlcek** [opensearch-project/documentation-website#399]

**Comment by hdhalter** [comment]
@rednaksi91  - Should this be moved to the Dashboards repo?

**Comment by hdhalter** [comment]
@opensearch-project/admin - can we move this to the appropriate repo? Thanks!

**Comment by bbarani** [comment]
@andrross @dblock I think this issue should be part of core? Do you agree?

**Comment by andrross** [comment]
@bbarani Yes this should be transferred to [core](https://github.com/opensearch-project/OpenSearch)

@lukas-vlcek I think the issue here is that only public Java classes are included in the Javadoc and the other classes you pointed to all have package private visibility. I spot checked a few other cases as well to confirm package private classes are not included in the Javadoc.
